File: application/app/Http/Requests/CreateExpenseRequest.php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;

class  CreateExpenseRequest extends FormRequest
{
    /**
     * Determine if the user is authorized to make this request.
     */
    public function authorize(): bool
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
     */
    public function rules(): array
    {
        return [
            'name'             => 'required|string|max:255|regex:/^[\pL\s\'.-]+$/u',
            'cost'             => 'required|numeric|regex:/^\d+(\.\d{1,2})?$/',
            'description'      => 'nullable|string|max:255|regex:/^[\pL\pN\s.,!?\-\'"]+$/u',
            'recurrenceRate'  => 'required|string|in:once,daily,weekly,monthly,yearly',
            'categoryId'      => 'nullable|integer',
            'nextDueDate'    => 'required|date_format:Y-m-d',
            'startDate'       => 'required|date_format:Y-m-d',
            'endDate'         => 'nullable|date_format:Y-m-d',
            'paidOnCreation' => 'required|boolean',
        ];
    }
}

File: server/BuilderServices/ExpenseService/Requests/CreateExpenseRequest.cs
namespace BuilderServices.ExpenseService.Requests;

public class CreateExpenseRequest
{
    public string Name { get; set; } = string.Empty;

    public double Cost { get; set; } = 0.0;

    public string? Description { get; set; } = string.Empty;

    public string RecurrenceRate { get; set; } = string.Empty;
    
    public string StartDate { get; set; } = string.Empty;

    public string? EndDate { get; set; } = string.Empty;

    public bool EndOfTheMonth { get; set; } = false;

    public int? CategoryId { get; set; } = 0;

    public bool PayToNow { get; set; } = false;

    public bool PayToNowIsCredit { get; set; } = false;
    
    public int? PayToNowCreditCardId { get; set; }

    public bool OneTimeExpenseIsPaid { get; set; } = false;
    
    public string? OneTimeExpensePaymentDate { get; set; }

    public bool OneTimeExpenseIsCredit { get; set; } = false;
    
    public int? OneTimeExpenseCreditCardId { get; set; }
    
    public bool IsAutomaticPayment { get; set; } = false;
    
    public int? AutomaticPaymentCreditCardId { get; set; }
}
